src/main/java/org/minions/devfund/damianvp/battleship/Ship.java

```diff
@@ -0,0 +1,231 @@
+                return false;
+            }
+            Long r2 = Math.round((oceanLength - row) * reducer + incrementer);
+            rowLimitB = row + r2.intValue();
+            Long c2 = Math.round((column + this.getLength()) * reducer + incrementer);
+            columnLimitB = column + this.getLength() - c2.intValue();
+        } else {
+            if (row + this.getLength() > oceanLength) {
+                return false;
+            }
+            Long r2 = Math.round((row + this.getLength()) * reducer + incrementer);
+            rowLimitB = row + this.getLength() - r2.intValue();
+            Long c2 = Math.round((oceanLength - column) * reducer + incrementer);
+            columnLimitB = column + c2.intValue();
+        }
+        Long r = Math.round(row * reducer + incrementer);
+        rowLimitA = row - r.intValue();
+        Long c1 = Math.round((column) * reducer + incrementer);
+        columnLimitA = column - c1.intValue();
+
+        for (int i = rowLimitA; i <= rowLimitB; i++) {
+            for (int j = columnLimitA; j <= columnLimitB; j++) {
+                if (!ocean.getShipArray()[i][j].getShipType().equals("empty")) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    /**
+     * This method places a ship in specific ocean.
+     * @param row int row position.
+     * @param column int column position.
+     * @param horizontal boolean, (true if ship is horizontal, false if ship is vertical).
+     * @param ocean Ocean type.
+     */
+    void placeShipAt(int row, int column, boolean horizontal, final Ocean ocean) {
+        this.bowRow = row;
+        this.bowColumn = column;
+        this.horizontal = horizontal;
+
+        if (this.horizontal) {
+            for (int i = column; i < column + this.getLength(); i++) {
+                ocean.getShipArray()[row][i] = this;
+            }
+        } else {
+            for (int j = row; j < row + this.getLength(); j++) {
+                ocean.getShipArray()[j][column] = this;
+            }
+        }
+    }
+
+    /**
+     * This method shot at in a specific position.
+     * @param row int row position.
+     * @param column int column position.
+     * @return true, If a part of the ship occupies the given row and column.
+     *         false, otherwise
+     */
+    boolean shootAt(int row, int column) {
+        if (!isSunk()) {
+            if (this.horizontal) {
+                if (row == this.bowRow && column >= this.bowColumn && column <= this.bowColumn + this.length) {
+                    this.hit[column - this.bowColumn] = true;
+                    return true;
+                }
+            } else {
+                if (column == this.bowColumn && row >= this.bowRow && row <= this.bowRow + this.length) {
+                    this.hit[row - this.bowRow] = true;
+                    return true;
+                }
+            }
+        }
+        return false;
```

Comment:
This may be refactored like this.
```java
    boolean shootAt(int row, int column) {
        if (isSunk()) {
            return false;
        }
        final boolean isHittable = this.horizontal ? row == this.bowRow && column >= this.bowColumn && column <= this.bowColumn + this.length
                : column == this.bowColumn && row >= this.bowRow && row <= this.bowRow + this.length;
        if (isHittable) {
            final int hitPosition = this.horizontal ? column - this.bowColumn : row - this.bowRow;
            this.hit[hitPosition] = true;
            return true;
        }
        return false;
    }
```

Reply:
yes thanks, is better